.github/workflows/cypress-tests-runner.yml

```diff
@@ -22,6 +22,45 @@ env:
   RUST_MIN_STACK: 10485760
 
 jobs:
+  formatter:
+    name: Run formatter on Cypress tests and address lints
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref || github.ref }}
+
+      - name: Install Node.js
+        if: ${{ env.RUN_TESTS == 'true' }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          cache: 'npm'
+
+      - name: Install Cypress and dependencies
+        if: ${{ env.RUN_TESTS == 'true' }}
+        run: |
+          npm ci --prefix ./cypress-tests
+
+      - name: Check formatting
+        if: ${{ env.RUN_TESTS == 'true' }}
+        shell: bash
+        run: |
+          npm run format --prefix cypress-tests
+          npm run lint --prefix cypress-tests -- --fix
+
+          if ! git diff --exit-code --quiet -- cypress-tests; then
+            echo "::notice::Cypress formatting and lint check failed"
+
+            git config --local user.name 'hyperswitch-bot[bot]'
+            git config --local user.email '148525504+hyperswitch-bot[bot]@users.noreply.github.com'
+
+            git add cypress-tests
+            git commit --message 'chore(cypress): run formatter and address lints'
+            git push
```

Comment:
Have you confirmed if the commit and push steps work correctly?

Reply:
Yes, this [commit](https://github.com/juspay/hyperswitch/pull/6588/commits/01f41c68b8477b848c788b6b164255c62cc0ab52) was actually done by that (from local, not from action).

Reply:
I'd prefer the steps be tested from the action as well.